pkgs/servers/aeron/default.nix

```diff
@@ -0,0 +1,168 @@
+}:
+
+let
+  pname = "aeron";
+  version = "1.40.0";
+
+  src = fetchFromGitHub {
+    owner = "real-logic";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-4C5YofA/wxwa7bfc6IqsDrw8CLQWKoVBCIe8Ec7ifAg=";
+  };
+
+  deps = stdenv.mkDerivation {
+    name = "${pname}-deps";
+    inherit src;
+
+    nativeBuildInputs = [
+      gradle
+      jdk11
+      perl
+    ];
+
+    buildPhase = ''
+      export GRADLE_USER_HOME=$(mktemp -d);
+      gradle \
+          --no-daemon \
+        build \
+          -Dorg.gradle.java.home="${jdk11.home}" \
+          -x test \
+          -x checkstyleMain \
+          -x checkstyleGenerated \
+          -x checkstyleGeneratedTest \
+          -x checkstyleMain \
+          -x checkstyleTest
+    '';
+
+    # Mavenize dependency paths
+    # e.g. org.codehaus.groovy/groovy/2.4.0/{hash}/groovy-2.4.0.jar -> org/codehaus/groovy/groovy/2.4.0/groovy-2.4.0.jar
+    installPhase = ''
+      find "$GRADLE_USER_HOME/caches/modules-2" -type f -regex '.*\.\(jar\|pom\)' \
+        | perl -pe 's#(.*/([^/]+)/([^/]+)/([^/]+)/[0-9a-f]{30,40}/([^/\s]+))$# ($x = $2) =~ tr|\.|/|; "install -Dm444 $1 \$out/$x/$3/$4/$5" #e' \
+        | sh
+      ln -s "$out/com/squareup/okio/okio/2.10.0/okio-jvm-2.10.0.jar" "$out/com/squareup/okio/okio/2.10.0/okio-2.10.0.jar"
+    '';
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "sha256-QRL3y7HF6/pWsC5MbtIqhZ4HR8jcWtQG6x9kajeoNMY=";
+  };
+
+  # Point to our local deps repo
+  gradleInit = writeText "init.gradle" ''
+    settingsEvaluated { settings ->
+      settings.pluginManagement {
+        repositories {
+          clear()
+          maven { url '${deps}' }
+        }
+      }
+    }
+    gradle.projectsLoaded {
+      rootProject.allprojects {
+        repositories {
+          clear()
+          maven { url '${deps}' }
+        }
+      }
+    }
+  '';
+
+  # replace buildSrc
+  buildSrc = writeText "build.gradle" ''
+    repositories {
+      clear()
+      maven { url '${deps}' }
+    }
+
+    dependencies {
+      implementation 'org.asciidoctor:asciidoctorj:2.5.5'
+      implementation 'org.eclipse.jgit:org.eclipse.jgit:5.13.1.202206130422-r'
+    }
+  '';
+
+in stdenv.mkDerivation rec {
+
+  inherit pname src version;
+
+  nativeBuildInputs = [
+    glib
+    gradle
+    jdk11
+    makeWrapper
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    export GRADLE_USER_HOME=$(mktemp -d)
+    cp ${buildSrc} ./buildSrc/build.gradle
+
+    gradle -PVERSION=${version} \
+        --offline \
+        --no-daemon \
+        --init-script "${gradleInit}" \
+      build \
+        -Dorg.gradle.java.home="${jdk11.home}" \
+        -x test \
+        -x checkstyleMain \
+        -x checkstyleGenerated \
+        -x checkstyleGeneratedTest \
+        -x checkstyleMain \
+        -x checkstyleTest
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D --mode=0444 --target-directory="$out/share/java" \
```

Comment:
Isn't that mode the default one?

Reply:
The default mode is 0755, at least according to install --help:

`  -m, --mode=MODE     set permission mode (as in chmod), instead of rwxr-xr-x`

Existing practice seems to be to set the mode on jar files to 0444:

```
nix-env -iA nixpkgs.javaPackages.jogl_2_3_2
ls -al ~/.nix-profile/share/java/jogl-all.jar
-r--r--r-- 2 root root 3401119 Jan  1  1970 /home/vaci/.nix-profile/share/java/jogl-all.jar
```

Reply:
normally not